src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.h

```diff
@@ -0,0 +1,111 @@
+    void operator=(PersistentStorageOperationalKeystoreHSM const &) = delete;
+
+    /**
+     * @brief Initialize the Operational Keystore to map to a given storage delegate.
+     *
+     * @param storage Pointer to persistent storage delegate to use. Must outlive this instance.
+     * @retval CHIP_NO_ERROR on success
+     * @retval CHIP_ERROR_INCORRECT_STATE if already initialized
+     */
+    CHIP_ERROR Init(PersistentStorageDelegate * storage)
+    {
+        VerifyOrReturnError(mStorage == nullptr, CHIP_ERROR_INCORRECT_STATE);
+        mPendingFabricIndex       = kUndefinedFabricIndex;
+        mIsExternallyOwnedKeypair = false;
+        mStorage                  = storage;
+        mPendingKeypair           = nullptr;
+        mIsPendingKeypairActive   = false;
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * @brief Finalize the keystore, so that subsequent operations fail
+     */
+    void Finish()
+    {
+        VerifyOrReturn(mStorage != nullptr);
+
+        ResetPendingKey();
+        mStorage = nullptr;
+    }
+
+    bool HasPendingOpKeypair() const override { return (mPendingKeypair != nullptr); }
+
+    bool HasOpKeypairForFabric(FabricIndex fabricIndex) const override;
+    CHIP_ERROR NewOpKeypairForFabric(FabricIndex fabricIndex, MutableByteSpan & outCertificateSigningRequest) override;
+    CHIP_ERROR ActivateOpKeypairForFabric(FabricIndex fabricIndex, const Crypto::P256PublicKey & nocPublicKey) override;
+    CHIP_ERROR CommitOpKeypairForFabric(FabricIndex fabricIndex) override;
+    CHIP_ERROR RemoveOpKeypairForFabric(FabricIndex fabricIndex) override;
+    void RevertPendingKeypair() override;
+    CHIP_ERROR SignWithOpKeypair(FabricIndex fabricIndex, const ByteSpan & message,
+                                 Crypto::P256ECDSASignature & outSignature) const override;
+    Crypto::P256Keypair * AllocateEphemeralKeypairForCASE() override;
+    void ReleaseEphemeralKeypair(Crypto::P256Keypair * keypair) override;
+
+protected:
+    void ResetPendingKey()
+    {
+        if (!mIsExternallyOwnedKeypair && (mPendingKeypair != nullptr))
+        {
+            Platform::Delete(mPendingKeypair);
+        }
+        mPendingKeypair           = nullptr;
+        mIsExternallyOwnedKeypair = false;
+        mIsPendingKeypairActive   = false;
+        mPendingFabricIndex       = kUndefinedFabricIndex;
+    }
+
+    PersistentStorageDelegate * mStorage = nullptr;
+
+    // This pending fabric index is `kUndefinedFabricIndex` if there isn't a pending keypair override for a given fabric.
+    FabricIndex mPendingFabricIndex          = kUndefinedFabricIndex;
+    Crypto::P256KeypairHSM * mPendingKeypair = nullptr;
+    bool mIsPendingKeypairActive             = false;
+
+    // If overridding NewOpKeypairForFabric method in a subclass, set this to true in
+    // `NewOpKeypairForFabric` if the mPendingKeypair should not be deleted when no longer in use.
+    bool mIsExternallyOwnedKeypair = false;
```

Comment:
Do you expect this to be used?

Or put another way, should this class be `final`?

Reply:
Removed "mIsExternallyOwnedKeypair"